tx.go

```diff
@@ -250,7 +251,12 @@ func (tx *Tx) close() {
 	} else {
 		tx.db.removeTx(tx)
 	}
+
+	// Clear all references.
```

Comment:
Could you put this stuff in a defer? Would it make sense? Perhaps the code is just clearer down here.